src/transformers/models/beit/image_processing_beit.py

```diff
@@ -236,6 +237,11 @@ def _preprocess_image(
         """Preprocesses a single image."""
         # All transformations expect numpy arrays.
         image = to_numpy_array(image)
+        if _is_scaled_image(image) and do_rescale:
```

Comment:
Do you think it makes sense to also set `do_rescale = False` ?

Reply:
I don't want to do anything "magic" and surprise the user. 

Checking whether the pixels are correct isn't a perfect test, and it'd be modifying the processor behaviour under an assumption.